udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/Step2AddDatasets.vue

```diff
@@ -0,0 +1,152 @@
+          class="w-100"
+          v-model="linkedDataset"
+          @change="getLinkedDataset"
+        />
+        <div v-if="datasetNotFound" class="fr-col bg-contrast-grey text-align-center fr-p-2v">
+          <p class="fr-text--md fr-text--bold">{{ t('No dataset associated to that link has been found') }}</p>
+        </div>
+      </fieldset>
+      <Alert type="error" v-if="errors.length" class="fr-mb-2w">
+        <template #title>{{ t("An error occured | Some errors occured", errors.length) }}</template>
+        <ul v-if="errors.length > 1">
+          <li v-for="error in errors">{{ error }}</li>
+        </ul>
+        <p v-else> {{ errors[0] }}</p>
+      </Alert>
+      <div class="fr-grid-row fr-grid-row--right">
+        <button class="fr-btn" @click="submit">
+          {{ t("Next") }}
+        </button>
+      </div>
+    </Container>
+</template>
+<script setup lang="ts">
+import { ref, toValue } from 'vue';
+import { useI18n } from 'vue-i18n';
+import Container from '../../components/Ui/Container/Container.vue';
+import InputGroup from '../../components/Form/InputGroup/InputGroup.vue';
+import MultiSelect from '../../components/MultiSelect/MultiSelect.vue';
+import Stepper from '../../components/Form/Stepper/Stepper.vue';
+import Alert from '../../components/Alert/Alert.vue';
+import useFunctionalState from '../../composables/form/useFunctionalState';
+import { requiredWithCustomMessage } from '../../i18n';
+import { api } from '../../plugins/api';
+import CardSm from '../../components/dataset/CardSM.vue';
+import type { Dataset } from '@etalab/data.gouv.fr-components';
+import { Reuse } from '../../types';
+import { useToast } from "../../composables/useToast";
+  
+const props = defineProps<{
+  errors: Array<string>,
+  steps: Array<string>,
+  loading?: Boolean,
+  reuse: Reuse,
+}>();
+
+const emit = defineEmits<{
+  (event: 'next', reuse: Reuse): void,
+}>();
+
+const { t } = useI18n();
+const { toast } = useToast();
+
+const datasets = ref<Dataset[]>([]);
+const reuse = ref(props.reuse);
+const linkedDataset = ref<string>("");
+const datasetNotFound = ref<Boolean>(false);
+
+const datasetRequired = requiredWithCustomMessage(t("At least one dataset is required."));
+
+const requiredRules = {
+  datasets: { datasetRequired },
+};
+
+const warningRules = {
+  datasets: { datasetRequired },
+};
+
+const { validateRequiredRules } = useFunctionalState({ datasets }, requiredRules, warningRules);
+
+const addDataset = async (datasetId: string) => {
+  const existingDataset = datasets.value.find(dataset => dataset.id === datasetId);
+  if (!existingDataset) {
+    let newDataset = await api.get('datasets/' + datasetId);
+    datasets.value.push(newDataset.data);
+  }
+};
+
+const removeDataset = (index: number) => datasets.value.splice(index, 1);
+
+const getLinkedDataset = async () => {
+  datasetNotFound.value = false;
+  const regex = /\/datasets\//;
+  
+  if (regex.test(linkedDataset.value)) {
```

Comment:
You can use [includes](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/String/includes) instead of a regex for this condition